### PR TITLE
chore: remove old Qt 5.12 code, update docs and FreeBSD runner

### PR DESCRIPTION
**Pull request by Nerixyz**
As the title says, this PR removes code disabled in builds with Qt 5.15.2 or later, updates docs (some minor stuff) and updates the FreeBSD runner to 14.0.

### CODE REVIEW

File: BUILDING_ON_WINDOWS_WITH_VCPKG.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ This will require more than 30GB of free space on your hard drive.
      See [VCPKG_ROOT documentation](https://learn.microsoft.com/en-gb/vcpkg/users/config-environment#vcpkg_root)
    - Append the vcpkg path to your path  
      e.g. `setx PATH "%PATH%;<path to vcpkg>"`
-   - For more configurations, see https://learn.microsoft.com/en-gb/vcpkg/users/config-environment
+   - For more configurations, see <https://learn.microsoft.com/en-gb/vcpkg/users/config-environment>
```
Comment (pajlada) [inline BUILDING_ON_WINDOWS_WITH_VCPKG.md:32]: what does this syntax do?

File: BUILDING_ON_FREEBSD.md [hunk 2]
```diff
@@ -1,25 +1,20 @@
...
-   ```
+   ports tree): `# pkg install boost-libs git qt6-base qt6-svg qtkeychain-qt6 cmake`
 1. Generate build files. To enable Lua plugins in your build add `-DCHATTERINO_PLUGINS=ON` to this command.
    ```sh
-   cmake ..
+   cmake -B build
    ```
 1. Build the project
    ```sh
-   make
```
Comment (pajlada) [inline BUILDING_ON_FREEBSD.md:24]: undo build-related changes

File: BUILDING_ON_FREEBSD.md [hunk 3]
```diff
@@ -1,25 +1,20 @@
...
-1. In the project directory, create a build directory and enter it
-   ```sh
-   mkdir build
-   cd build
-   ```
+   ports tree): `# pkg install boost-libs git qt6-base qt6-svg qtkeychain-qt6 cmake`
 1. Generate build files. To enable Lua plugins in your build add `-DCHATTERINO_PLUGINS=ON` to this command.
    ```sh
-   cmake ..
+   cmake -B build
```
Comment (pajlada) [inline BUILDING_ON_FREEBSD.md:15]: Add the `-DBUILD_WITH_QT6=ON` flag

File: BUILDING_ON_WINDOWS.md [hunk 4]
```diff
@@ -33,8 +33,7 @@ When prompted which components to install, do the following:
 1. Unfold the tree element that says "Qt"
 2. Unfold the top most tree element (latest stable Qt version, e.g. `Qt 6.5.3`)
 3. Under this version, select the following entries:
-   - `MSVC 2019 64-bit` (or alternative version if you are using that)
-   - `Qt 5 Compatibility Module`
+   - `MSVC 2019 64-bit` (or `MSVC 2022 64-bit` from Qt 6.8 onwards)
```
Comment (pajlada) [inline BUILDING_ON_WINDOWS.md:36]: What made us not need the compat module on Windows anymore?

File: BUILDING_ON_WINDOWS.md [hunk 5]
```diff
@@ -66,9 +65,9 @@ These dependencies are only required if you are not using a package manager
    - Visit the downloads list on [SourceForge](https://sourceforge.net/projects/boost/files/boost-binaries/).
    - Select the latest version from the list.
    - Download the `.exe` file appropriate to your Visual Studio installation version and system bitness (choose `-64` for 64-bit systems).
-     Visual Studio versions map as follows: `14.3` in the filename corresponds to MSVC 2022,`14.2` to 2019, `14.1` to 2017, `14.0` to 2015. _Anything prior to Visual Studio 2015 is unsupported. Please upgrade should you have an older installation._
+     Visual Studio versions map as follows: `14.3` in the filename corresponds to MSVC 2022. _Anything prior to Visual Studio 2022 is unsupported. Please upgrade should you have an older installation._
```
Comment (pajlada) [inline BUILDING_ON_WINDOWS.md:68]: Keep the 2019 version, I'm happy to try to support if should someone be stuck with it

File: BUILDING_ON_WINDOWS_WITH_VCPKG.md (hunk 1 above)
Comment (Nerixyz) [reply BUILDING_ON_WINDOWS_WITH_VCPKG.md:32]: Although GitHub Markdown recognizes the link either way, [CommonMark](https://spec.commonmark.org/0.31.2/#autolinks) requires the link to be wrapped in `<>`. This basically fixes [MD034](https://github.com/DavidAnson/markdownlint/blob/v0.34.0/doc/md034.md).

Comment (pajlada) [reply BUILDING_ON_WINDOWS_WITH_VCPKG.md:32]: kk that's fine then :+1:

File: .cirrus.yml [hunk 6]
```diff
@@ -1,5 +1,5 @@
 freebsd_instance:
-  image: freebsd-14-0-release-amd64
+  image_family: freebsd-14-0
```
Comment (pajlada) [inline .cirrus.yml:2]: nice

**Review by pajlada — APPROVED** [review]

**Pull request merged**